src/api/app/models/bs_request.rb

```diff
@@ -76,7 +76,7 @@ class BsRequest < ApplicationRecord
 
   has_many :bs_request_actions, dependent: :destroy
   has_many :reviews, dependent: :delete_all
-  has_many :comments, as: :commentable, dependent: :delete_all
+  has_many :comments, as: :commentable, dependent: :destroy
```

Comment:
Is this change related to the PR?

Reply:
Well, it's somehow related and we discovered this when testing different use cases in notifications. But I could maybe open a new PR just for that and extend the description. 

I tried to perform all the possible changes that can happen to a request (create one, add a review on it, change to different states, delete it) in order to check what kind of notification was created on each case.

When I did delete the request (something that only Admins via API can do) I realized that existing notifications for comments on that request were not deleted so we got an error when we tried to access the notifications page.

Reply:
New PR open for that: #10735